File: jan2/jan2.rb
def is_palindrome(str)
    formatted_str = str.downcase.gsub(/[^a-zA-Z0-9]/, '')
    reversed_str = formatted_str.reverse
    formatted_str == reversed_str
end

input = "A man, a plan, a canal, Panama"
result = is_palindrome(input)
puts result

File: jan2/jan2.java
public class Palindrome {
    public static boolean isPalindrome(String str) {
        String formattedStr = str.toLowerCase().replaceAll("[^a-zA-Z0-9]", "");
        String reversedStr = new StringBuilder(formattedStr).reverse().toString();
        return formattedStr.equals(reversedStr);
    }

    public static void main(String[] args) {
        String input = "A man, a plan, a canal, Panama";
        boolean result = isPalindrome(input);
        System.out.println(result);
    }
}
